include/wasm.hh

```diff
@@ -648,8 +648,8 @@ public:
   Func() = delete;
   ~Func();
 
-  using callback = auto (*)(const Val[], Val[]) -> own<Trap>;
-  using callback_with_env = auto (*)(void*, const Val[], Val[]) -> own<Trap>;
+  using callback = auto (*)(const vec<Val>&, vec<Val>&) -> own<Trap>;
```

Comment:
Just FYI, allocating the param/result data structures is highly performance sensitive for fast calls. For maximum optimization, it's important that an engine can materialize the respective objects "by hand", i.e. precisely predict the C++ object layout. That's trivially doable with a `Val[]` and should still be fine with a `const vec<Val>&`, so I'm just pointing this out in case further changes will be discussed/considered here.
(FWIW, we haven't gone that far yet in V8, we're still using `malloc` here, which is fairly slow. C++-side stack allocation of dynamically sized arrays is not an option for us.)